Vegetarian-Food-Finder/Controller/SearchViewController.swift

```diff
@@ -6,46 +6,67 @@
+final class SearchViewController: UIViewController, UITableViewDataSource, UITableViewDelegate, UISearchBarDelegate, UISearchResultsUpdating {
     private let searchBarCell = "SearchBarCell"
     private let segueFromSearchToDetails = "SegueFromSearchToDetails"
     @IBOutlet var tableView: UITableView!
     let searchController = UISearchController()
-    var restaurantsList: [Business] = []
+    var restaurantList: [Business] = []
+    var filteredRestaurantsList: [Business] = []
+    var activeRestaurantsList: [Business] {
+        searchController.isActive ? filteredRestaurantsList : restaurantList
+    }
+
     override func viewDidLoad() {
         super.viewDidLoad()
         title = "Search"
         setUpSearchController()
         tableView.dataSource = self
         tableView.delegate = self
         tableView.rowHeight = UITableView.automaticDimension
-        setUpSearchController()
         tableView.reloadData()
     }
 
     func setUpSearchController() {
+        searchController.searchResultsUpdater = self
         searchController.obscuresBackgroundDuringPresentation = false
         searchController.searchBar.enablesReturnKeyAutomatically = false
         searchController.searchBar.placeholder = "Search"
         definesPresentationContext = true
         navigationItem.searchController = searchController
         navigationItem.hidesSearchBarWhenScrolling = false
-        searchController.searchBar.scopeButtonTitles = ["All", "Near By", "Top Rated", "Affordable"]
         searchController.searchBar.delegate = self
     }
 
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return restaurantsList.count
+        return activeRestaurantsList.count
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: searchBarCell, for: indexPath) as! SearchBarCell
-        let thisRestaurant = restaurantsList[indexPath.row]
+        let thisRestaurant = activeRestaurantsList[indexPath.row]
         cell.configure(for: thisRestaurant)
         return cell
     }
 
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         performSegue(withIdentifier: segueFromSearchToDetails, sender: self)
     }
+
+    func updateSearchResults(for searchController: UISearchController) {
+        let searchBar = searchController.searchBar
+        let searchText = searchBar.text!
+        filterForTextSearch(searchText: searchText)
+    }
+
+    func filterForTextSearch(searchText: String) {
+        filteredRestaurantsList = restaurantList.filter({ Business in
+            if searchController.searchBar.text != "" {
+                let searchTextMatch = Business.name.lowercased().contains(searchText.lowercased())
+                return searchTextMatch
+            }
+            return true
+        })
+        tableView.reloadData()
```

Comment:
This function is taking care of two different things, filtering data and reloading the table. This may be better somewhere else. Like where the text changes